pyhms/demes/single_pop_eas/multiwinner.py

```diff
@@ -0,0 +1,190 @@
+from typing import Callable, Protocol
+
+import numpy as np
+
+from ...core.population import Population
+from ...core.problem import Problem
+from .common import VariationalOperator
+
+
+class UtilityFunction:
+    """
+    Utility function for multiwinner voting.
+
+    For more details, please refer to these papers:
+    1. Faliszewski, Piotr, et al. "Multiwinner voting in genetic algorithms for
+    solving ill-posed global optimization problems."
+    """
+
+    def __init__(self, distance: Callable, gamma: Callable, delta: Callable) -> None:
+        self.distance = distance
+        self.gamma = gamma
+        self.delta = delta
+
+    def evaluate(
+        self,
+        voter_genome: np.ndarray,
+        candidate_genome: np.ndarray,
+        candidate_fitness: float,
+        problem: Problem,
+    ) -> float:
+        self.h = (lambda x: x) if problem.maximize else (lambda x: 1 / (1 + x))
+        reversed_fitness = self.h(candidate_fitness)
+        distance_value = self.distance(voter_genome, candidate_genome)
+        return self.gamma(reversed_fitness) * self.delta(distance_value)
```

Comment:
We can encounter function with scaled domain and different fitness ranges. Shouldn't the population fitnesses and genomes be normalized so that the proportion of their impact on the metric is preserved?

Reply:
Then we could also use a simple `h = 1 - x` for minimization.

Reply:
I agree, we should introduce some sort of normalization.

Reply:
I would like to discuss it on Wednesday. Right now I will merge it without normalization.